mp/src/game/shared/shareddefs.h

```diff
@@ -85,15 +85,19 @@ class CViewVectors
 #define VEC_DEAD_VIEWHEIGHT	g_pGameRules->GetViewVectors()->m_vDeadViewHeight
 
 #ifdef NEO
-#define NEO_RECON_VIEW_OFFSET Vector(0.0f, 0.0f, -6.0f)
-#define NEO_ASSAULT_VIEW_OFFSET Vector(0.0f, 0.0f, -3.75f)
-#define NEO_SUPPORT_VIEW_OFFSET Vector(0.0f, 0.0f, 3.0f)
+#define NEO_RECON_VIEW_OFFSET Vector(0.0f, 0.0f, 0.0f)
+#define NEO_ASSAULT_VIEW_OFFSET Vector(0.0f, 0.0f, -0.5f)
+#define NEO_SUPPORT_VIEW_OFFSET Vector(0.0f, 0.0f, 0.0f)
+#define NEO_RECON_DUCKED_VIEW_OFFSET Vector(0.0f, 0.0f, -1.96f)
+#define NEO_ASSAULT_DUCKED_VIEW_OFFSET Vector(0.0f, 0.0f, -2.96f)
+#define NEO_SUPPORT_DUCKED_VIEW_OFFSET Vector(0.0f, 0.0f, 3.05f)
+
 #define VEC_VIEW_NEOSCALE(NeoPlayer) (g_pGameRules->GetViewVectors()->m_vView + ((NeoPlayer->GetClass() == NEO_CLASS_RECON) ? NEO_RECON_VIEW_OFFSET : (NeoPlayer->GetClass() == NEO_CLASS_SUPPORT) ? NEO_SUPPORT_VIEW_OFFSET : NEO_ASSAULT_VIEW_OFFSET))
-#define VEC_DUCK_VIEW_NEOSCALE(NeoPlayer) (g_pGameRules->GetViewVectors()->m_vDuckView + ((NeoPlayer->GetClass() == NEO_CLASS_RECON) ? NEO_RECON_VIEW_OFFSET : (NeoPlayer->GetClass() == NEO_CLASS_SUPPORT) ? NEO_SUPPORT_VIEW_OFFSET : NEO_ASSAULT_VIEW_OFFSET))
+#define VEC_DUCK_VIEW_NEOSCALE(NeoPlayer) (g_pGameRules->GetViewVectors()->m_vDuckView + ((NeoPlayer->GetClass() == NEO_CLASS_RECON) ? NEO_RECON_DUCKED_VIEW_OFFSET : (NeoPlayer->GetClass() == NEO_CLASS_SUPPORT) ? NEO_SUPPORT_DUCKED_VIEW_OFFSET : NEO_ASSAULT_DUCKED_VIEW_OFFSET))
 
-#define NEO_RECON_DUCK_MAXHULL_OFFSET Vector(0.0f, 0.0f, -2.0f)
-#define NEO_ASSAULT_DUCK_MAXHULL_OFFSET Vector(0.0f, 0.0f, -1.0f)
-#define NEO_SUPPORT_DUCK_MAXHULL_OFFSET Vector(0.0f, 0.0f, 6.0f)
+#define NEO_RECON_DUCK_MAXHULL_OFFSET Vector(0.0f, 0.0f, -2.161614f) // Hull Max of 59 when crouched
+#define NEO_ASSAULT_DUCK_MAXHULL_OFFSET Vector(0.0f, 0.0f, -0.826738f) // Hull Max of 48 when crouched
+#define NEO_SUPPORT_DUCK_MAXHULL_OFFSET Vector(0.0f, 0.0f, 6.244769f) // Hull Max of 46 when crouched
```

Comment:
Where do these Z values come from? A fraction of standing hull vs ducked hulls maxs.z?

Reply:
I don't know what the exact hull maxes in og are, but there is a phenomenon where the player vphysics shadow moves up from the ground slightly after moving (present in base sdk2013 and ce too). These values give the most headroom to the collision box so that the player can still fit into the same hole when right against it and when moving for some time as in the original to the nearest hammer unit, while not being able to fit in the next largest hole even in the worst case scenario (to the nearest hammer unit). the actual offset applied to the hull max will be further modified by the player scale which I have included in the readout on the right side of the screen

Reply:
I don't know how this increase in the collision box altitude works exactly but it looks like it raises the collision box by about half a hammer unit, so the values can probably be moved by no more than that to find some nice round number. There was the crouch jitter which i was also trying to negate by giving as much headroom as possible while keeping the same hull max to within one hammer unit, but that has since been fixed in a different pr so maybe we can change these values to something more round

Reply:
@Rainyan Are we confident the current hull widths are parity? If that is the case we can probably use a sloped roof with cl_showpos to figure out the actual height of the playermodel to a far greater degree of accuracy

Reply:
![image](https://github.com/NeotokyoRebuild/neo/assets/44210793/6f6c5e91-ed2f-408f-a3f3-0a8ad29db713)
So I guess that's a no. Probably the current x and y hull dimensions do not take the PlayerScale into account or something

Reply:
Fractions make more sense now, gonna close this

Reply:
> @Rainyan Are we confident the current hull widths are parity?

I believe we are off by one, at least for some classes. You can test this using the "funnel" area in the dimensions_test map against the OG NT.